ocaml/testsuite/tests/typing-local/regions.ml

```diff
@@ -18,6 +18,37 @@ let check_empty name =
 
 let () = check_empty "startup"
 
+let foo x = local_
+  ignore_local (Some x);
+  ()
+
+let foo_bad : 'a -> unit = Obj.magic foo
+
+external follow : int -> ('a [@local_opt]) = "%int_as_pointer"
+
+let[@inline never] int_as_pointer_local x =
+  foo_bad x;
+  (* The region should be preserved by the following call; hence no stack space
+     leaking *)
+  let _ = opaque_identity (follow 0) in
```

Comment:
I think using `0` here means getting a pointer that might explode in no-naked-pointers mode. You're probably going to need to write a C stub that creates a genuine external pointer.

Reply:
I wrote a C stub and it seems to be working. However, I didn't use the ocaml5-proof approach that you mentioned because I forgot.